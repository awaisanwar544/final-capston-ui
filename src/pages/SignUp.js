import { useDispatch } from 'react-redux';
import logo from '../assets/logo.png';

import { signUp } from '../redux/reducers/user';

function SignUp() {
  const dispatch = useDispatch();

  const handleSubmit = (event) => {
    event.preventDefault();
    const name = event.target[0].value;
    const email = event.target[1].value;
    const password = event.target[2].value;
    dispatch(signUp(name, email, password));
  };
  return (
    <div className="w-screen h-screen flex items-center bg-custom-green-500">
      <form className="bg-white shadow-xl rounded px-8 pt-6 pb-8 h-fit mx-auto max-w-xs" onSubmit={handleSubmit}>
        <div className="flex w-full p-10 justify-center">
          <a href="/">
            <img className="w-24" src={logo} alt="logo" />
          </a>
        </div>
        <div className="mb-4">
          <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="username">
            Username
            <input className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="username" type="text" placeholder="Username" />
          </label>
        </div>
        <div className="mb-4">
          <label className="block text-gray-700 text-sm font-bold mb-4" htmlFor="email">
            E-mail
            <input type="email" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="email" placeholder="Email@something.com" />
          </label>
        </div>
        <div className="mb-6">
          <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="password">
            Password
            <input className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="password" type="password" placeholder="******************" />
          </label>
        </div>
        <div className="flex items-center justify-center">
          <button className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none" type="submit">
            Sign Up
          </button>
        </div>
      </form>
    </div>
  );
}

export default SignUp;
