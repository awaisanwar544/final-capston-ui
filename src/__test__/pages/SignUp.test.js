import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Routes, Route } from 'react-router';
import store from '../../redux/store';
import SignUp from '../../pages/SignUp';

describe('App', () => {
  test('Render App', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" exact element={<SignUp />} />
          </Routes>
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
