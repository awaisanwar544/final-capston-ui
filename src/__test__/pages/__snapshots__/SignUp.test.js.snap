// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render App 1`] = `
<div
  className="w-screen h-screen flex flex-col items-center justify-center space-y-10 bg-custom-green-500"
>
  <form
    className="bg-white shadow-xl rounded px-8 pt-6 pb-8 h-fit mx-auto max-w-xs"
    onSubmit={[Function]}
  >
    <div
      className="flex w-full p-10 justify-center"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          className="w-24"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-2"
        htmlFor="username"
      >
        Username
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="username"
          placeholder="Username"
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-4"
        htmlFor="email"
      >
        E-mail
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="email"
          placeholder="Email@something.com"
          required={true}
          type="email"
        />
      </label>
    </div>
    <div
      className="mb-6"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-2"
        htmlFor="password"
      >
        Password
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="password"
          placeholder="******************"
          required={true}
          type="password"
        />
      </label>
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-2"
        htmlFor="confirm-password"
      >
        Confirm password
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="confirm-password"
          placeholder="******************"
          required={true}
          type="password"
        />
      </label>
    </div>
    <div
      className="flex items-center justify-center"
    >
      <button
        className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none"
        type="submit"
      >
        Sign Up
      </button>
    </div>
    <a
      className="inline-block mt-10 align-baseline font-bold text-sm text-custom-grey-500 hover:text-custom-green-500"
      exact="true"
      href="/signin"
      onClick={[Function]}
    >
      Already have an account?
      <br />
      Sign in now!
    </a>
  </form>
</div>
`;
